tests/templates/test_subroutines/test_qubitization.py

```diff
@@ -0,0 +1,274 @@
+
+
+@pytest.mark.parametrize(
+    "hamiltonian, expected_unitaries",
+    (
+        (
+            qml.ops.LinearCombination(
+                np.array([1, -1, 2]), [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+            ),
+            [
+                qml.PauliX(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+                qml.PauliY(0) @ qml.GlobalPhase(np.array(np.pi), wires=0),
+                qml.PauliZ(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+            ],
+        ),
+        (
+            qml.ops.LinearCombination(
+                np.array([1.0, 1.0, 2.0]), [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+            ),
+            [
+                qml.PauliX(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+                qml.PauliY(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+                qml.PauliZ(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+            ],
+        ),
+        (
+            qml.ops.LinearCombination(
+                np.array([-0.2, -0.6, 2.1]), [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+            ),
+            [
+                qml.PauliX(0) @ qml.GlobalPhase(np.array(np.pi), wires=0),
+                qml.PauliY(0) @ qml.GlobalPhase(np.array(np.pi), wires=0),
+                qml.PauliZ(0) @ qml.GlobalPhase(np.array(0), wires=0),
+            ],
+        ),
+    ),
+)
+def test_positive_coeffs_hamiltonian(hamiltonian, expected_unitaries):
+    """Tests that the function _positive_coeffs_hamiltonian correctly transforms the Hamiltonian"""
+
+    new_coeffs, new_unitaries = _positive_coeffs_hamiltonian(hamiltonian)
+
+    assert np.allclose(new_coeffs, np.abs(hamiltonian.terms()[0]))
+
+    for i, unitary in enumerate(new_unitaries):
+        assert qml.equal(expected_unitaries[i], unitary)
+
+
+def test_template_definition():
+    """Tests that the Qubitization template is correctly defined.
+    Based on eq.(65): https://arxiv.org/abs/2204.11890"""
+
+    H = qml.dot([0.1, 0.3, -0.3], [qml.Z(0), qml.Z(1), qml.Z(0) @ qml.Z(2)])
+    lambda_ = sum([abs(term) for term in H.terms()[0]])
+
+    diag_H = np.diag(qml.matrix(H))
+    diag_Q = np.diag(
+        qml.matrix(qml.Qubitization(H, control=[3, 4]), wire_order=[3, 4, 0, 1, 2])[:8, :8]
+    )
+    assert np.allclose(diag_H, diag_Q * lambda_)
+
+
+@pytest.mark.usefixtures("use_legacy_and_new_opmath")
+def test_legacy_new_opmath():
+    coeffs, ops = [0.1, -0.3, -0.3], [qml.X(0), qml.Z(1), qml.Y(0) @ qml.Z(2)]
+
+    H1 = qml.dot(coeffs, ops)
+    matrix_H1 = qml.matrix(qml.Qubitization(H1, control=[3, 4]), wire_order=[3, 4, 0, 1, 2])
+
+    H2 = qml.Hamiltonian(coeffs, ops)
+    matrix_H2 = qml.matrix(qml.Qubitization(H2, control=[3, 4]), wire_order=[3, 4, 0, 1, 2])
+
+    assert np.allclose(matrix_H1, matrix_H2)
+
+
+@pytest.mark.xfail(reason="see https://github.com/PennyLaneAI/pennylane/issues/5507")
+@pytest.mark.usefixtures("use_legacy_and_new_opmath")
+def test_legacy_new_opmath_diff():
```

Comment:
Why this is not inside TestDifferentiability?

Reply:
You are right 👍